### PR TITLE
Add flags

### PR DESCRIPTION
**Pull request by ynqa**
### Overview

- Add flags for:
  - `Verbose` to ignore unnecessary stdouts.
  - `DType` to select gorgonia dtype from float32|float64.
  - `Profile` to run with pprof.

### CODE REVIEW

File: builder/word2vec.go [hunk 1]
```diff
@@ -75,24 +82,6 @@ func NewWord2VecBuilderViper() *Word2VecBuilder {
...
-			"Optimizer:          %v\n"+
-			"MaxDepth:           %v\n"+
-			"NegativeSampleSize: %v\n"+
-			"Theta:              %v\n"+
-			"BatchSize:          %v\n"+
-			"SubsampleThreshold: %v\n",
-		wb.dimension, wb.window, wb.initLearningRate,
-		wb.toLower, wb.model, wb.optimizer, wb.maxDepth, wb.negativeSampleSize,
-		wb.theta, wb.batchSize, wb.subsampleThreshold)
-}
```
Comment (ynqa) [inline builder/word2vec.go]: No displaying the summary of configs.

File: model/dtype.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/chewxy/gorgonia/tensor"
+	"github.com/pkg/errors"
+)
+
+// Dtype stores gorgonia dtype and engine.
+type Dtype struct {
+	T tensor.Dtype
+	E tensor.Engine
+}
```
Comment (ynqa) [inline model/dtype.go]: What is good for naming?

**Comment by ynqa** [comment]
`state.go` is more complicated than before...

**Comment by ynqa** [comment]
merge to `master` directly

File: model/dtype.go (hunk 2 above)
Comment (chewxy) [reply model/dtype.go]: call it `Type` because you're combining a `Dtype` with its execution engine

Comment (ynqa) [reply model/dtype.go]: Sure

**Comment by chewxy** [comment]
LGTM.

One note on potential optimization (not necessary, but good to keep in mind): instead of `*model.Type` being passes around, why not just `model.Type`. This reduces the chance that your code may be accidentally overwriting the `Dtype` or `Engine`

**Comment by ynqa** [comment]
I got it, thanks.

**Pull request merged**